packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/settings/PluginSettingsConfigurable.kt

```diff
@@ -27,21 +28,22 @@ class PluginSettingsConfigurable : Configurable {
     }
 
     override fun isModified(): Boolean {
-        val savedSettings = useSettings()
+        val savedSettings by service<PluginSettingsStateComponent>()
         return settingsComponent.isTelemetryEnabledCheckBox.isSelected !=
-            savedSettings.isTelemetryEnabled
+            savedSettings.state.isTelemetryEnabled
     }
 
     override fun apply() {
-        val savedSettings =
-            useSettings().apply {
-                isTelemetryEnabled = settingsComponent.isTelemetryEnabledCheckBox.isSelected
-            }
+        val savedSettings by service<PluginSettingsStateComponent>()
+        savedSettings.state.apply {
```

Comment:
yea so this is one such side effect, that what we're exposing as service is an implementation specifics. If the service was not fetching PluginSettingsStateComponent but lets say "PluginSettings" class with interface such as `setTelemetryEnabled`, etc then we won't need to worry about digging `.state` out of savedSettings :D

Reply:
yea that sounds good 👍